tests/dsp_testing/filters.ipynb

```diff
@@ -0,0 +1,947 @@
+    "print('\\n************** Equivalent single-stage filter **************')\n",
+    "print(equivalent_new_filter)\n",
+    "\n",
+    "fig, ax = plt.subplots(1, 1)\n",
+    "phase_ax = ax.twinx()\n",
+    "plot_stage_freq_response_on_ax(fig, ax, equivalent_new_filter.stages[0], equivalent_new_filter.input_rates[0])\n",
+    "plot_stage_phase_response_on_ax(fig, phase_ax, equivalent_new_filter.stages[0], equivalent_new_filter.input_rates[0], color='tab:orange')"
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "id": "80f2220e-a805-4fea-b764-c6742b2b5339",
+   "metadata": {},
+   "source": [
+    "We can see some copies of the second filter stage in near the central passband. The first filter stage only suppresses side lobes by 115 dB, so the closest two copies on either side are above the noise floor. Is this acceptable? It is up to the filter designer and the requirements of the filter."
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "id": "f7925744-b5ec-4469-ad48-70b0f58961c0",
+   "metadata": {},
+   "source": [
+    "# Testing on real data\n",
+    "Included in this directory is a single sequence of data taken from a `rawrf` file. The radar was transmitting at 10.5 MHz at this time, with a center frequency of 12.0 MHz. First, we load in the data, then plot the time and frequency domain representations to gain some insight on the spectrum."
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": null,
+   "id": "b9b4b4f2-af88-4724-b779-7d21eea42581",
+   "metadata": {},
+   "outputs": [],
+   "source": [
+    "example_data = np.load(\"example_sequence.npz\")\n",
+    "sequence_samples_cpu = example_data['sqn_data']\n",
+    "# The sequence is prepended with data before the start of the first pulse, assuming the default DecimationScheme was to be used. \n",
+    "default_extra_samples = 0\n",
+    "taps = [stage.filter_taps for stage in default_scheme.stages]\n",
+    "for dm, taps in zip(reversed(default_scheme.dm_rates), reversed(taps)):\n",
+    "    default_extra_samples = (default_extra_samples * dm) + len(taps) // 2\n",
+    "print('Extra samples: ', default_extra_samples)\n",
+    "\n",
+    "_ = plot_sequence(sequence_samples_cpu, sample_rate, 'Sequence Data')\n",
+    "# plot_sequence(sequence_samples_cpu, sample_rate, 'Sequence Data', freq_lim=(-1.75e6, -1.25e6))"
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "id": "10bf5bae-b2dc-42b5-9847-8958ca180e73",
+   "metadata": {},
+   "source": [
+    "From the frequency domain representation, it is clear that there are a lot of significant signals within the band. The transmitted signal is at an offset frequency of -1.5 MHz, and is far from the strongest signal. This is a good test of how well the filtering schemes can remove unwanted interference from other sources in the band.\n",
+    "\n",
+    "## Default filtering scheme\n",
+    "First, let's test out the default filtering scheme that Borealis uses."
+   ]
+  },
+  {
+   "cell_type": "code",
```

Comment:
The plots starting here and the sections after are all formatted far to large. I can only see about half of the plot, while the right half of it extends off screen.